model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -257,6 +262,141 @@ message ProcessBundleProgressRequest {
   string instruction_reference = 1;
 }
 
+message MonitoringInfo {
+  // The name defining the metric or monitored state.
+  string urn = 1;
+
+  // This is specified as a URN that implies:
+  // A message class: (Distribution, Counter, Extrema, MonitoringDataTable).
+  // Sub types like field formats - int64, double, string.
+  // Aggregation methods - SUM, LATEST, TOP-N, BOTTOM-N, DISTRIBUTION
+  // valid values are:
+  // beam:metrics:[SumInt64|LatestInt64|Top-NInt64|Bottom-NInt64|
+  //     SumDouble|LatestDouble|Top-NDouble|Bottom-NDouble|DistributionInt64|
+  //     DistributionDouble|MonitoringDataTable]
+  string type = 2;
+
+  // The Metric or monitored state.
+  oneof monitoring_status {
+    MonitoringTableData monitored_table_data = 3;
+    Metric metric = 4;
+  }
+
+  // A set of key+value labels which define the scope of the metric.
+  // Either a well defined entity id for the keys:
+  // “transform”, “pcollection”, “windowing_strategy”,
```

Comment:
Want to add an enum defining these "well" known strings?

This will allow developers across languages to have a consistent spelling.